core/Test/Tasty/Core.hs

```diff
@@ -173,8 +174,12 @@ data Progress = Progress
   }
   deriving
   ( Show -- ^ @since 1.2
+  , Eq   -- ^ @since 1.6
   )
 
+emptyProgress :: Progress
+emptyProgress = Progress mempty 0.0
```

Comment:
Add `@since 1.5` and a haddock comment please. OTOH do we really need to export this entity? Given that users are restricted to test ingredients / providers, it seems not that bad to spell `Progress { progressText = ..., progressPercent = ... }` explicitly.

Reply:
This term is only used outside of this module, and it's quite useful not just to construct progress but also to check if the progress is non empty.